src/extension/background-script/connectors/commando.ts

```diff
@@ -0,0 +1,334 @@
+};
+type CommandoListInvoiceResponse = {
+  invoices: CommandoInvoice[];
+};
+
+type CommandoInvoice = {
+  label: string;
+  status: string;
+  description: string;
+  amount_received_msat: number;
+  bolt11: string;
+  payment_preimage: string;
+  paid_at: number;
+  payment_hash: string;
+};
+export default class Commando implements Connector {
+  config: Config;
+  ln: LnMessage;
+
+  constructor(config: Config) {
+    this.config = config;
+    this.ln = new LnMessage({
+      remoteNodePublicKey: this.config.pubkey,
+      wsProxy: this.config.wsProxy || "wss://lnwsproxy.regtest.getalby.com",
+      ip: this.config.host,
+      port: this.config.port || 9735,
+      privateKey: this.config.privateKey,
+      // logger: {
+      //   info: console.log,
+      //   warn: console.warn,
+      //   error: console.error
+      // },
+    });
+  }
+
+  async init() {
+    // initiate the connection to the remote node
+    await this.ln.connect();
+  }
+
+  async unload() {
+    await this.ln.disconnect();
+  }
+
+  async connectPeer(
+    args: ConnectPeerArgs
+  ): Promise<ConnectPeerResponse | Error> {
+    return this.ln
+      .commando({
+        method: "connect",
+        params: {
+          id: args.pubkey,
+          host: args.host,
+        },
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        return {
+          data: true,
+        };
+      })
+      .catch((err) => {
+        return new Error(err);
+      });
+  }
+
+  async getInvoices(): Promise<GetInvoicesResponse> {
+    return this.ln
+      .commando({
+        method: "listinvoices",
+        params: {},
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoListInvoicesResponse;
+        return {
+          data: {
+            invoices: parsed.invoices.map(
+              (invoice, index): ConnectorInvoice => ({
+                id: invoice.label,
+                memo: invoice.description,
+                settled: invoice.status == "paid",
+                preimage: invoice.payment_preimage,
+                settleDate: invoice.paid_at,
+                type: "received",
+                totalAmount: (invoice.amount_received_msat / 1000).toString(),
+              })
+            ),
+          },
+        };
+      });
+  }
+
+  async getInfo(): Promise<GetInfoResponse> {
+    const response = (await this.ln.commando({
+      method: "getinfo",
+      params: {},
+      rune: this.config.rune,
+    })) as CommandoGetInfoResponse;
+    return {
+      data: {
+        alias: response.alias,
+        pubkey: response.id,
+        color: response.color,
+      },
+    };
+  }
+
+  async getBalance(): Promise<GetBalanceResponse> {
+    const response = (await this.ln.commando({
+      method: "listfunds",
+      params: {},
+      rune: this.config.rune,
+    })) as CommandoListFundsResponse;
+    let lnBalance = 0;
+    for (let i = 0; i < response.channels.length; i++) {
```

Comment:
nit: Something like this might be easier to read:
```js
response.channels.forEach(channel => lnBalance += channel.channel_sat);
```

Reply:
could probably be also nicely written with `reduce()`

```
lnBalance = response.channels.reduce((balance, channel) => balance + channel.channel_sat, 0)
```

Reply:
:D reduce was my first thought, but then... I hate to look at reduce :P